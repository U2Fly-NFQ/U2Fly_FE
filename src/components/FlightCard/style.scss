@use '../../assets/scss/abstracts/variables' as var;

.flight-card {
  margin-bottom: 36px;

  .flight-card-items {
    background: #ffffff;
    border: 1px solid #dddddd;
    border-radius: 12px;
    display: flex;
    justify-content: space-between;
    align-items: center;

    @media screen and (max-width: 768px) {
      flex-direction: column;
      align-items: unset;
    }
  }

  .flight-card-place {
    display: flex;
    justify-content: space-between;
    max-width: 620px;

    @media screen and (max-width: 576px) {
      flex-direction: column;
    }

    @media screen and (max-width: 972px) {
      padding: 16px 0;
    }
  }

  .flight-card-place__airline,
  .flight-card-place__arrow {
    @media screen and (max-width: 972px) {
      padding-left: 25px;
      display: none;
    }
  }

  .flight-card-place__arrow h6 {
    font-size: 11px;
    font-weight: 700;
    padding-top: 10px;
  }

  .flight-card-place__arrow p {
    font-size: 11px;
    line-height: 18px;
  }

  .flight-card-place__left,
  .flight-card-place__right {
    display: flex;
    align-items: center;
    padding-left: 23px;
    @media screen and (max-width: 972px) {
      padding-left: 0;
      align-items: unset;
      flex-direction: column;
    }
  }

  .flight-card-place-destination {
    padding-left: 33px;
    @media screen and (max-width: 992px) {
      padding-left: 25px;
    }

    &__sub-title,
    &__title,
    &__decs {
      margin: 0;
      padding: 0;
    }

    &__sub-title {
      font-size: 14px;
      line-height: 16px;
      color: var.$paragraph-color;
    }

    &__title {
      font-weight: 500;
      padding-top: 5px;
      font-size: 24px;
    }

    &__desc {
      padding-top: 5px;
      font-size: 12px;
      width: 140px;
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
      color: var.$dark;
    }
  }

  .flight-card-price {
    background: #f3eff9;
    padding: 25px 16px 20px 25px;

    .flight-card-price__origin {
      color: var.$paragraph-color;
      font-size: 18px;
      font-weight: 500;
    }

    .flight-card-price__off {
      padding-top: 10px;
      font-size: 36px;
      font-weight: 500;
      line-height: 40px;

      sup {
        font-size: 13px;
        color: var.$main-color;
        font-weight: 600;
        top: -19px;
        padding-left: 5px;
      }
    }

    &__condition {
      font-size: 12px;
      padding-top: 2px;
      font-weight: 500;
      color: var(--black-color);
    }

    &__show-more {
      font-size: 16px;
      font-weight: 500;
      padding-top: 7px;
      cursor: pointer;
    }
  }
}
