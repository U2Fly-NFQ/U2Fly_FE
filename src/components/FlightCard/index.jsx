import React from 'react'
import { Skeleton } from 'antd'
import { motion } from 'framer-motion'
import './style.scss'
import { useNavigate } from 'react-router-dom'
import { getLsObj, updateLs } from '../../utils/localStorage'
import { useTranslation } from 'react-i18next'
import moment from 'moment'

Number.prototype.round = function (places) {
  return +(Math.round(this + 'e+' + places) + 'e-' + places)
}

export default function FlightCard(props) {
  const { data, loading } = props
  const navigate = useNavigate()
  const { t } = useTranslation()

  const onBooking = () => {
    let flight = getLsObj('flight')

    if (flight.ticketType) {
      if (flight.ticketType === 'oneWay') {
        updateLs('flight', {
          id: data.id,
        })
        navigate('/flights-booking')
        return
      }

      if (flight.ticketType === 'roundTrip') {
        if (flight.id) {
          updateLs('flight', {
            roundId: data.id,
          })
          navigate('/flights-booking')
        } else {
          updateLs('flight', {
            id: data.id,
          })
          navigate(0)
        }
        return
      }
    }
  }

  if (loading)
    return (
      <Skeleton
        loading={true}
        active
        paragraph={{
          rows: 6,
        }}
      ></Skeleton>
    )

  const randomDiscount = Math.random().toFixed(2)

  return (
    <motion.div
      whileHover={{
        scale: 1.01,
        boxShadow: '0 1rem 3rem rgba(#000, 0.175)',
      }}
    >
      <div className="flight-card">
        <div className="place">
          <div className="airline">
            <img
              className="airline-image"
              src="https://andit.co/projects/html/and-tour/assets/img/common/biman_bangla.png"
            />
          </div>

          <div className="destination">
            <p className="sub-title">{t('flight-list-page.From')}</p>
            <h3 className="title">
              {data.departure.city} ({data.departure.iata})
            </h3>
            <h6 className="desc">
              {moment(data.startTime, 'HH:mm:ss').format('HH:mm')}
            </h6>
          </div>
          <div className="arrow">
            <h6>{t('flight-list-page.Direct')}</h6>
            <i className="fa-solid fa-arrow-right-long"></i>
            <p>
              {data.duration} {t('flight-list-page.hour')}{' '}
            </p>
          </div>
          <div className="destination">
            <p className="sub-title">{t('flight-list-page.To')}</p>
            <h3 className="title">
              {data.arrival.city} ({data.arrival.iata})
            </h3>
            <h6 className="desc">
              {moment(data.startTime, 'HH:mm:ss')
                .add(data.duration * 60, 'minutes')
                .format('HH:mm')}
            </h6>
          </div>
        </div>
        <div className="flight-card-deal">
          <div className="flight-card-price">
            <h5 className="flight-card-price__origin">
              <del>${data.seat.price}</del>
            </h5>
            <h2 className="flight-card-price__discount">
              ${(data.seat.price * (1 - randomDiscount)).toFixed(2)}
            </h2>
            <h4 className="discount-label red">
              <span>
                {t('flight-list-page.OFF', {
                  number: (randomDiscount * 100).toFixed(0),
                })}
              </span>
            </h4>
          </div>
          <button className="btn btn-primary btn-md" onClick={onBooking}>
            Book now
          </button>
        </div>
      </div>
    </motion.div>
  )
}
