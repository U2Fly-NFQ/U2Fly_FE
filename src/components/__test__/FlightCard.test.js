import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import FlightCard from '../FlightCard'

const mockStore = configureStore([])

describe('My Connected React-Redux Component', () => {
  let store
  let component
  beforeEach(() => {
    let data = {
      status: 'success',
      flight: {
        id: 1,
        code: 'VN156',
        arrival: {
          id: 2,
          iata: 'HAN',
          name: 'Noi Bai International Airport',
          city: 'Ha Noi',
          image: 'google.com',
        },
        departure: {
          id: 3,
          iata: 'VCA',
          name: 'Can Tho International Airport',
          city: 'Can Tho',
          image: 'google.com',
        },
        startTime: '07:00:00',
        startDate: '2022-07-09',
        duration: 1.5,
        airplane: {
          id: 1,
          name: 'Boeing 787',
        },
        airline: {
          id: 1,
          name: 'Vietnam Airlines',
          icao: 'HVN',
          image:
            'https://www.vietnamairlines.com/~/media/Images/VNANew/Home/Logo%20Header/logo_vna-mobile.png',
        },
        seat: {
          id: 1,
          name: 'Economy',
          price: 59,
          seatAvailable: 49,
          discount: 0.1,
        },
      },
    }
    component = renderer.create(
      //   <Provider store={store}>
      <Router>
        <FlightCard loading={data.status} data={data.flight} />
      </Router>
      //   </Provider>
    )
  })
  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })
})
