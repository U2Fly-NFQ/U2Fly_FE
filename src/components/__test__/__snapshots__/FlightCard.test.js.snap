// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<div
  className="ant-skeleton ant-skeleton-active"
>
  <div
    className="ant-skeleton-content"
  >
    <h3
      className="ant-skeleton-title"
      style={
        Object {
          "width": "38%",
        }
      }
    />
    <ul
      className="ant-skeleton-paragraph"
    >
      <li
        style={
          Object {
            "width": undefined,
          }
        }
      />
      <li
        style={
          Object {
            "width": undefined,
          }
        }
      />
      <li
        style={
          Object {
            "width": undefined,
          }
        }
      />
      <li
        style={
          Object {
            "width": undefined,
          }
        }
      />
      <li
        style={
          Object {
            "width": undefined,
          }
        }
      />
      <li
        style={
          Object {
            "width": "61%",
          }
        }
      />
    </ul>
  </div>
</div>
`;
